syntax_def_dev.py

```diff
@@ -508,3 +508,77 @@ def inhibit(ret):
         ]
 
         return inhibit(self.base_completions + completions)
+
+class SyntaxDefRegexCaptureGroupHighlighter(sublime_plugin.ViewEventListener):
+    @classmethod
+    def is_applicable(cls, settings):
+        syntax = settings.get('syntax')
+        return syntax == 'Packages/PackageDev/Package/Sublime Text Syntax Definition/Sublime Text Syntax Definition.sublime-syntax'
+
+    def on_selection_modified(self):
+        self.view.erase_regions('captures')
+        self.view.add_regions('captures', list(self.get_regex_regions()), 'comment')
```

Comment:
you could use settings to get the user's preferred scope name and styles - again, just like the syntax test highlighting does.